src/modules/SX126x.cpp

```diff
@@ -1058,6 +1063,29 @@ int16_t SX126x::setFrequencyRaw(float freq) {
   return(ERR_NONE);
 }
 
+int16_t SX126x::setDio2AsRfSwitch(bool enable) {
+  int dio2 = _mod->getInt1();
+  if (dio2 == -1)
+  {
+    // DIO2 is not defined, return error
+    _mod->setDio2Func(false);
+    return ERR_WRONG_MODEM;
+  }
+  uint8_t* data = new uint8_t[1];
```

Comment:
Memory leak. There's no reason to allocate dynamically, use 
```suggestion
  uint8_t data[1];
```